Bitbucket-Pipelines/two-stage-pipeline-template/{{cookiecutter.outputDir}}/bitbucket-pipelines.yml

```diff
@@ -0,0 +1,56 @@
+# Prerequisites: $AWS_ACCESS_KEY_ID and $AWS_SECRET_ACCESS_KEY setup in the Deployment variables https://support.atlassian.com/bitbucket-cloud/docs/set-up-and-monitor-deployments/.
+
+image: atlassian/default-image:2
+pipelines:
+  default:
+    - step:
+        name: Test
+        script:
+          - echo "Tests here..."
+          # Add testing logic here
+
+  branches:
+    {{ cookiecutter.main_git_branch }}:
+      - parallel:
+        - step:
+            name: Test
+            script:
+              - echo "Tests here..."
+              # Add testing logic here
+        - step:
+            name: Security Scan
+            script:
+              # Run a security scan for sensitive data.
+              # See more security tools at https://bitbucket.org/product/features/pipelines/integrations?&category=security
+              - pipe: atlassian/git-secrets-scan:0.5.1
+      - step:
+          name: Deploy to Testing
+          deployment: Testing
+          script:
+            - pipe: atlassian/aws-sam-deploy:1.3.0
+              variables:
+                AWS_ACCESS_KEY_ID: $AWS_ACCESS_KEY_ID
+                AWS_SECRET_ACCESS_KEY: $AWS_SECRET_ACCESS_KEY
```

Comment:
Hi @okyrdan,

I did some testing on this, this approach won't work. The credentials configured is usually to assume a role in individual accounts.

Please check our implementation for [Gitlab where we assume role via a scrip](https://github.com/aws/aws-sam-cli-pipeline-init-templates/blob/main/Gitlab/two-stage-pipeline-template/%7B%7Bcookiecutter.outputDir%7D%7D/.gitlab-ci.yml#L75) or [Github where assume-role is supported by default](https://github.com/aws/aws-sam-cli-pipeline-init-templates/blob/main/GitHub-Actions/two-stage-pipeline-template/%7B%7Bcookiecutter.outputDir%7D%7D/.github/workflows/pipeline.yaml#L102). Please let me know if you need more help to implement this.

Reply:
Hi @c2tarun,
It works only for exact user in current implementation.
Yep, it's a good case to add a support for assuming role. Working on it.

Reply:
Hi @c2tarun,
Thanks for you help.
We are planning to work on this feature soon. We'll notify you additionally )
Keep in touch!

Reply:
Hi @orkydan, thank you for this contribution! We want to include this PR into an upcoming release on Nov 3 because it aligns with the roadmap. To meet the internal deadlines, we can help addressing the feedback mentioned by @c2tarun. Do you think you’ll be able to complete this by 10/27? If this is unlikely, we’d be happy to tweak it. Please let us know. Again, thank you for this!